File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.java
class KthLargest {
    int size;
    PriorityQueue<Integer>pq;
    public KthLargest(int k, int[] nums) {
        size = k;
        pq = new PriorityQueue<>();
        for(int x:nums) {
            pq.add(x);
            if(pq.size() > k) pq.poll();
        }
    }
    
    public int add(int val) {
        pq.add(val);
        if(pq.size() > size) pq.poll();
        return pq.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */

File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.cpp
class KthLargest {
public:
    int size;
    priority_queue<int>pq;
    KthLargest(int k, vector<int>& nums) {
        size = k;
        for(int x:nums) {
            pq.push(-x);
            if(pq.size() > size) pq.pop();
        }
    }
    
    int add(int val) {
        pq.push(-val);
        if(pq.size() > size) pq.pop();
        return -1 * pq.top();
    }
};

/*

class KthLargest {
public:
    int size;
    priority_queue<int, vector<int>, greater<int>>pq;
    KthLargest(int k, vector<int>& nums) {
        size = k;
        for(int x:nums) {
            pq.push(x);
            if(pq.size() > size) pq.pop();
        }
    }
    
    int add(int val) {
        pq.push(val);
        if(pq.size() > size) pq.pop();
        return pq.top();
    }
};


*/
/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */